openhcl/underhill_core/src/worker.rs

```diff
@@ -1863,43 +1872,67 @@ async fn new_underhill_vm(
         crate::inspect_proc::periodic_telemetry_task(driver_source.simple()),
     );
 
+    let dma_pool = if let Some(shared_vis_pages_pool) = shared_vis_pages_pool.clone() {
```

Comment:
We cannot allow cloning the pools, for the reasons stated above. Which means, you'll need to refactor DmaManager to either take ownership of the pool, or have some method it can use to spawn allocators.

Reply:
You can probably temporarily get away with just taking the pool spawner to allow you to spawn allocators, but you cannot allow multiple instances of the pool around because then usage / allocation is broken (multiple users of the same GPA range)